src/lib.rs

```diff
@@ -423,6 +423,56 @@ impl Build {
                 configure.arg("-D__STDC_NO_ATOMICS__");
             }
 
+            if target.contains("wasi") {
+                configure.args([
+                    // Termios isn't available whatsoever on WASM/WASI so we disable that
+                    // fatal error: 'termios.h' file not found
+                    "no-ui-console",
+                    // WASI doesn't support UNIX sockets so we preemptively disable it
+                    "no-sock",
+                    // WASI doesn't have a concept of syslog, so we disable it
+                    // fatal error: 'syslog.h' file not found
+                    "-DNO_SYSLOG",
+                    // wasm lacks signal support; to enable minimal signal emulation, compile with
+                    // -D_WASI_EMULATED_SIGNAL and link with -lwasi-emulated-signal
+                    "-D_WASI_EMULATED_SIGNAL",
+                    // WASI lacks process-associated clocks; to enable emulation of the `times` function using the wall
+                    // clock, which isn't sensitive to whether the program is running or suspended, compile with
+                    // -D_WASI_EMULATED_PROCESS_CLOCKS and link with -lwasi-emulated-process-clocks
+                    "-D_WASI_EMULATED_PROCESS_CLOCKS",
+                    // WASI lacks a true mmap; to enable minimal mmap emulation, compile
+                    // with -D_WASI_EMULATED_MMAN and link with -lwasi-emulated-mman
+                    "-D_WASI_EMULATED_MMAN",
+                    // WASI lacks process identifiers; to enable emulation of the `getpid` function using a
+                    // placeholder value, which doesn't reflect the host PID of the program, compile with
+                    // -D_WASI_EMULATED_GETPID and link with -lwasi-emulated-getpid
+                    "-D_WASI_EMULATED_GETPID",
+                    // ----------
+                    // The following arguments are taken from:
+                    // https://github.com/wapm-packages/OpenSSL/blob/master/build.sh#L21
+                    // Since this is a successful and reliable source for configuring WASM/I builds.
+                    // I will try to explain those to the best of my ability though.
+                    // ----------
```

Comment:
Personally I think it's best to only list here what we need and explain the reason for adding them, so I think it's ok to remove this reference since each flag below should have comments as to why it's there.

Reply:
Done!